core/templates/domain/exploration/curated-exploration-validation.service.ts

```diff
@@ -0,0 +1,143 @@
+import { MultipleChoiceInputCustomizationArgsBackendDict } from
+  'extensions/interactions/customization-args-defs';
+import constants from 'assets/constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CuratedExplorationValidationService {
+  constructor(
+    private explorationSummaryBackendApiService:
+      ExplorationSummaryBackendApiService,
+    private readOnlyExplorationBackendApiService:
+      ReadOnlyExplorationBackendApiService) {}
+
+  /* *****************************************
+  Validate exploration settings.
+  ***************************************** */
+
+  // Exploration must be published.
+  async isExpPublishedAsync(explorationId: string): Promise<boolean> {
+    return this.explorationSummaryBackendApiService.
+      loadPublicExplorationSummariesAsync([explorationId]).then(
+        (response: ExplorationSummaryBackendDict) => {
+          let summaries = response.summaries;
+          return (summaries.length === 1 && summaries[0] !== null);
+        });
+  }
+
+  // Exploration must have correctness feedback enabled.
+  async isCorrectnessFeedbackEnabled(explorationId: string): Promise<boolean> {
+    return this.readOnlyExplorationBackendApiService
+      .fetchExplorationAsync(explorationId, null).then(
+        (response: FetchExplorationBackendResponse) => {
+          return response.correctness_feedback_enabled;
+        });
+  }
+
+  // Exploration cannot be in a custom category.
+  async isDefaultCategoryAsync(explorationId: string): Promise<boolean> {
+    return this.explorationSummaryBackendApiService
+      .loadPublicExplorationSummariesAsync([explorationId]).then(
+        (response: ExplorationSummaryBackendDict) => {
+          let summaries = response.summaries;
+          let isCategoryPresent = false;
+          if (summaries.length === 1) {
+            let category = summaries[0].category;
+            for (let i of constants.ALL_CATEGORIES) {
+              if (i === category) {
+                isCategoryPresent = true;
+              }
+            }
+          }
+          return isCategoryPresent;
+        });
+  }
+
+  /* *****************************************
+  Validate exploration RTE components.
+  ***************************************** */
+
+  // We perform all RTE validation checks in the frontend by making a request
+  // to the server and returning the result of the backend validation check.
+  // This allows us to avoid needing to parse HTML DOM strings.
+
+  /* *****************************************
+  Validate exploration interactions.
+  ***************************************** */
+
+  // Interactions are restricted to a whitelist.
+  ALLOWED_INTERACTIONS = [
+    'Continue', 'EndExploration', 'MultipleChoiceInput', 'ItemSelectionInput',
+    'FractionInput', 'NumericInput', 'TextInput', 'DragAndDropSortInput',
+    'ImageClickInput', 'RatioExpressionInput', 'NumericExpressionInput',
+    'AlgebraicExpressionInput', 'MathEquationInput'];
```

Comment:
Can we move this to the common constants file?

Reply:
Turns out there is a constant for it already! Thanks for the feedback.